### DISCUSSION TITLE
ForeignKey to Model not yet Defined

### DISCUSSION THREAD

**Discussion by cancan101** [sqlalchemy/sqlalchemy#6292] [Usage Questions]
Other than passing a string to `ForeignKey`, is there anyway to point a `ForeignKey` to a model not yet defined, e.g. using a `lambda`? [`relationships`](https://docs.sqlalchemy.org/en/13/orm/relationship_api.html#sqlalchemy.orm.relationship.params.argument) do allow passing a callable.
```python
class ModelA(Base):
   # this works: 
   model_b_id = Column(ForeignKey("model_b.id"), index=True)

   # this does not, but is nicer for various IDEs: 
   model_b_id = Column(ForeignKey(lambda: ModelB.id), index=True)

class ModelB(Base):
   id = ....
```

**Comment by zzzeek** [comment]
not right now, this may have been requested a long time ago, not sure.
